components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/MGMRpcOpsImpl.kt

```diff
@@ -87,7 +91,18 @@ class MGMRpcOpsImpl @Activate constructor(
 
     private inner class ActiveImpl : InnerMGMRpcOps {
         override fun generateGroupPolicy(holdingIdentityShortHash: String): String {
-            return mgmOpsClient.generateGroupPolicy(ShortHash.ofOrThrow(holdingIdentityShortHash))
+            return try {
+                mgmOpsClient.generateGroupPolicy(ShortHash.ofOrThrow(holdingIdentityShortHash))
+            } catch (e: CouldNotFindMemberException) {
+                throw ResourceNotFoundException("Could not find $holdingIdentityShortHash")
+            } catch (e: MemberNotAnMgmException) {
+                throw InvalidInputDataException(
+                    details = mapOf(
+                        "holdingIdentityShortHash" to holdingIdentityShortHash
+                    ),
+                    message = "Member: $holdingIdentityShortHash is not an MGM!",
```

Comment:
Nit picky suggestion:
```suggestion
                    message = "Member with holding identity $holdingIdentityShortHash is not an MGM.",
```